test/unit/org/apache/cassandra/index/sai/cql/types/DataSet.java

```diff
@@ -0,0 +1,707 @@
+    {
+        @Override
+        BigDecimal[] emptyValues()
+        {
+            return new BigDecimal[NUMBER_OF_VALUES];
+        }
+
+        @Override
+        BigDecimal nextValue()
+        {
+            return getRandom().nextBigDecimal(-1_000_000, 1_000_000, -64, 64);
+        }
+
+        @Override
+        BigDecimal negate(BigDecimal value)
+        {
+            return value.signum() > 0 ? value.negate() : value;
+        }
+
+        @Override
+        BigDecimal abs(BigDecimal value)
+        {
+            return value.signum() < 0 ? value.abs() : value;
+        }
+
+        @Override
+        BigDecimal increment(BigDecimal value)
+        {
+            return value.add(BigDecimal.ONE);
+        }
+
+        public String toString()
+        {
+            return "decimal";
+        }
+    }
+
+
+    public static class FloatDataSet extends NumericDataSet<Float>
+    {
+        @Override
+        Float[] emptyValues()
+        {
+            return new Float[NUMBER_OF_VALUES];
+        }
+
+        @Override
+        Float nextValue()
+        {
+            return getRandom().nextFloat();
+        }
+
+        @Override
+        Float negate(Float value)
+        {
+            return value > 0 ? -value : value;
+        }
+
+        @Override
+        Float abs(Float value)
+        {
+            return value < 0 ? Math.abs(value) : value;
+        }
+
+        @Override
+        Float increment(Float value)
+        {
+            return ++value;
+        }
+
+        public String toString()
+        {
+            return "float";
+        }
+    }
+
+    public static class DoubleDataSet extends NumericDataSet<Double>
+    {
+        @Override
+        Double[] emptyValues()
+        {
+            return new Double[NUMBER_OF_VALUES];
+        }
+
+        @Override
+        Double nextValue()
+        {
+            return getRandom().nextDouble();
+        }
+
+        @Override
+        Double negate(Double value)
+        {
+            return value > 0 ? -value : value;
+        }
+
+        @Override
+        Double abs(Double value)
+        {
+            return value < 0 ? Math.abs(value) : value;
+        }
+
+        @Override
+        Double increment(Double value)
+        {
+            return ++value;
+        }
+
+        public String toString()
+        {
+            return "double";
+        }
+    }
+
+    public static class AsciiDataSet extends DataSet<String>
+    {
+        public AsciiDataSet()
+        {
+            values = new String[NUMBER_OF_VALUES];
+            List<String> list = Arrays.asList(values);
+            for (int index = 0; index < values.length; index++)
+            {
+                String value;
+                while (true)
+                {
+                    value = getRandom().nextAsciiString(8, 256);
+                    if (!list.contains(value))
+                        break;
+                }
```

Comment:
We could do:
```
do
{
    value = getRandom().nextAsciiString(8, 256);
} while (list.contains(value));
```
If we prefer this format, we can apply it to other `while(true)` calls in the file.